### DISCUSSION TITLE
[NodeJS][Bug] Failed to send heartbeat 14 UNAVAILABLE: TCP Read failed

### DISCUSSION THREAD

**Discussion by TonyKingdom** [apache/skywalking#8231] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

NodeJS Server Side Agent (apache/skywalking-nodejs)

### What happened

node version: v10.24.1
agent version: v0.3.0
base docker image version: 3.10.9

NodeJS App started in container with following error:

{ file:
   '/usr/lib/node_modules/skywalking-backend-js/lib/agent/protocol/grpc/clients/HeartbeatClient.js',
  code: 14,
  metadata: Metadata { _internal_repr: {}, flags: 0 },
  details: 'TCP Read failed',
  level: 'error',
  message: 'Failed to send heartbeat 14 UNAVAILABLE: TCP Read failed',
  stack:
   'Error: 14 UNAVAILABLE: TCP Read failed\n    at Object.exports.createStatusError (/usr/lib/node_modules/skywalking-backend-js/node_modules/grpc/src/common.js:91:15)\n    at Object.onReceiveStatus (/usr/lib/node_modules/skywalking-backend-js/node_modules/grpc/src/client_interceptors.js:1209:28)\n    at InterceptingListener._callNext (/usr/lib/node_modules/skywalking-backend-js/node_modules/grpc/src/client_interceptors.js:568:42)\n    at InterceptingListener.onReceiveStatus (/usr/lib/node_modules/skywalking-backend-js/node_modules/grpc/src/client_interceptors.js:618:8)\n    at callback (/usr/lib/node_modules/skywalking-backend-js/node_modules/grpc/src/client_interceptors.js:847:24)' }

### What you expected to happen

no idea what's going wrong.

### How to reproduce

Start agent as README described with follow software versions.

node version: v10.24.1
agent version: v0.3.0
base docker image version: 3.10.9

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
This seems you set to wrong port.

**Comment by TonyKingdom** [comment]
You mean collector port?
The OAP server address was set via environment variables as describe in README.

SW_AGENT_COLLECTOR_BACKEND_SERVICES | The backend OAP server address | 127.0.0.1:11800

I double checked the key and value and it was absolutely correct.

**Reply by wu-sheng** [reply]
I can't diagnose for your own env, but our e2e testing of nodejs agent is checking for every commit.
